src/ExtractTileOperations.cpp

```diff
@@ -143,6 +164,138 @@ Tile<3> get_3d_tile_index(const Expr &e) {
     return {true, base, {x_stride, 0, r_stride}, {x_tile, y_tile, r_tile}};
 }
 
+Tile<3> get_3d_rhs_tile_index(const Expr &e, int element_width) {
```

Comment:
Perhaps not necessary for this PR, but we really should express these as pattern-based rules, they would be far easier to parse and debug

Reply:
Yes, some of it is a bit messy. It would also be nice to have the ability to wildcard on the number of lanes so that arbitrary `Ramp`s and `Broadcast`s could be matched and have their number of lanes retrieved.

Reply:
Ah true, that would be quite convenient, perhaps such support could be added to the pattern matcher in a future PR.

Reply:
In comment form, could you add a fairly complete list of example Exprs that would match? That will help whoever changes it to pattern matching in the future.

Reply:
I think this is the last remaining comment request. Once you push a new commit, the buildbots should get themselves unconfused.

Reply:
oops, almost didn't see this, should be in 314656cbb1084923ace01b727623124f2dc5ee3b.
I remember the reason why I didn't use patterns for this is because I can't match on a wildcarded number of lanes. And any attempt to use `.with_lanes(0)` would make asserts in the `make` functions for the nodes fail since they often require at least 1 lane, but 0 is required to make `expr_match` ignore the amount of lanes when matching.

Reply:
Sure, but please add example patterns *as a comment* to help future readers